pkg/controller/configmap/configmap.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019 Hewlett Packard Enterprise Development LP
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	kdv1 "github.com/bluek8s/kubedirector/pkg/apis/kubedirector/v1beta1"
+	"github.com/bluek8s/kubedirector/pkg/observer"
+	"github.com/bluek8s/kubedirector/pkg/shared"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	// StatusGens is exported so that the validator can have access
+	// to the ConfigMap StatusGens
+	StatusGens = shared.NewStatusGens()
+)
+
+// syncConfigMap runs the reconciliation logic. It is invoked because of a
+// change in or addition of configmap instance, currently there is no
+// polling for this resource. If the configmap is not labeled
+// with key "kubedirectorcmtype" then no op
+func (r *ReconcileConfigMap) syncConfigMap(
+	reqLogger logr.Logger,
+	cr *corev1.ConfigMap,
+) error {
+	// Memoize state of the incoming object.
+	oldMap, _ := observer.GetConfigMap(cr.Namespace, cr.Name)
+
+	// Set a defer func to write new status and/or finalizers if they change.
+	defer func() {
+
+		if _, ok := oldMap.Labels["kubedirectorcmtype"]; !ok {
+			return
+		}
+		/* anonymous fun to check if some cluster
+		   is using this config map as a connection */
+		isClusterUsingConfigMap := func(cmName string, cluster kdv1.KubeDirectorCluster) bool {
+			clusterModels := cluster.Spec.Connections.ConfigMaps
+			for _, modelMapName := range clusterModels {
+				if modelMapName == cmName {
+					return true
+				}
+			}
+			return false
+		}
+		allClusters := &kdv1.KubeDirectorClusterList{}
+		shared.List(context.TODO(), allClusters)
+		for _, kubecluster := range allClusters.Items {
+			if isClusterUsingConfigMap(cr.Name, kubecluster) {
+				shared.LogInfof(
+					reqLogger,
+					cr,
+					shared.EventReasonConfigMap,
+					"configmap {%s} is connected to cluster {%s} updating its configmeta",
+					cr.Name,
+					kubecluster.Name,
+				)
```

Comment:
similar comments/questions about direction of connection and where to log the event

Reply:
Changed language slightly and also added log in the kubecluster